composer.json

```diff
@@ -22,6 +22,7 @@
     ],
     "require": {
         "php": "^7.1",
+        "doctrine/persistence": "^1.3.4 || ^2.0",
```

Comment:
i think we should check the dependencies of orm and raise the minimum version to one that has the refactoring to persistence, rather than having this line. otherwise it would be possible to install with an ancient orm, we'd install persistence in addition, but the orm would return the old objects and our code explodes.

Reply:
I'll change to https://github.com/doctrine/phpcr-odm/releases/tag/1.5.2, but I think we should still add `doctrine/persistence` since it is used in the package and I would also move some of the dev requirements to the `require` section as they are direct dependencies if you agree (in another PR).

Reply:
not sure what the best practice is. in the end, we rely on the orm package to not do BC breaks and return the expected class (that happens to be in the doctrine/persistence namespace). i don't think explicitly depending on doctrine/persistence is more robust or reliable - if doctrine does not return what we expect, things still won't work.

Reply:
the problem with that is that `doctrine/phpcr-odm` could allow `doctrine/persistence` 2 (as it happened) in a minor version and that makes tests fail, if there was a constraint to not be able to use `doctrine/persistence` `2.0`, it would work. https://travis-ci.org/github/doctrine/DoctrinePHPCRBundle/builds/733019189 should fail.

Reply:
that was a BC break in phpcr-odm - it made phpcr-odm change the classes of some objects it returns. (persistence 1 had the classes extend the legacy classes to avoid the BC break). when the code was written, the separate persistence layer did not exist yet. 

having this line does not protect us against other accidental BC breaks of any upstream libraries (apart from an hypothetical incompatible persistence 3 that phpcr-odm would suddenly allow).

so i am not sure. if you insist, we can keep the line, but i would rather remove it.

Reply:
what we need to do either way is declare a conflict with phpcr-odm < 1.5 and up the require-dev to use phpcr-odm 1.5

version 1.4 is not using doctrine/persistence yet if i understood it correctly. then the objects returned would not be the persistence objects at all.

Reply:
oh, phpcr-odm 1.4 seems to end up with a version of doctrine common that includes persistence 1 and does the BC thing, so it happens to work.